tracer/test/Datadog.Trace.Security.IntegrationTests/AspNetCore5AsmToggle.cs

```diff
@@ -58,6 +69,34 @@ public async Task TestSecurityToggling(bool? enableSecurity)
 
             await VerifySpans(spans.ToImmutableList(), settings, true);
         }
+
+        [SkippableFact]
+        [Trait("RunOnWindows", "True")]
+        public async Task TestRemoteConfigError()
+        {
+            var enableSecurity = true;
+            var url = "/Health/?[$slice]=value";
+            var agent = await RunOnSelfHosted(enableSecurity);
+            var settings = VerifyHelper.GetSpanVerifierSettings();
+
+            var spans1 = await SendRequestsAsync(agent, url);
+
+            agent.SetupRcm(Output, new[] { ((object)"haha, you weren't expect this!", "1") }, "FEATURES");
+
+            var request = await agent.WaitRcmRequestAndReturnLast();
+            CheckAckState(request, ApplyStates.ERROR, "Error converting value \"haha, you weren't expect this!\" to type 'Datadog.Trace.Configuration.Features'. Path '', line 1, position 32.", "First RCM call");
+
+            await VerifySpans(spans1.ToImmutableList(), settings, true);
+        }
+
+        private void CheckAckState(GetRcmRequest request, uint expectedState, string expectedError, string message)
+        {
+            var state = request?.Client?.State?.ConfigStates?.FirstOrDefault(x => x.Product == "FEATURES");
```

Comment:
```suggestion
            var state = request?.Client?.State?.ConfigStates?.SingleOrDefault(x => x.Product == "FEATURES");
```

Reply:
Will add to https://github.com/DataDog/dd-trace-dotnet/pull/3186